tokio/src/sync/once_cell.rs

```diff
@@ -1,13 +1,20 @@
-use super::{Semaphore, SemaphorePermit, TryAcquireError};
 use crate::loom::cell::UnsafeCell;
+use crate::loom::sync::atomic::AtomicU8;
+use crate::loom::sync::Mutex;
+use crate::util::linked_list::LinkedList;
+use crate::util::{linked_list, WakeList};
+use std::convert::Infallible;
 use std::error::Error;
 use std::fmt;
 use std::future::Future;
+use std::marker::PhantomPinned;
 use std::mem::MaybeUninit;
 use std::ops::Drop;
+use std::pin::Pin;
 use std::ptr;
-use std::sync::atomic::{AtomicBool, Ordering};
```

Comment:
The fact that this used to use atomics without going through `crate::loom` tells me that the `OnceCell` is not loom-tested at all. An implementation as complex as this will require the addition of [loom tests](https://github.com/tokio-rs/tokio/tree/master/tokio/src/sync/tests).

Reply:
I'm working on it right now, and found a non-trivial synchronization issue that I'm fixing. Will push the tests soon.

Reply:
It should be good now. As usual, thanks to `loom` for reminding me how hard correct synchronization is.